week-2-wide-perspective/w02e05-simple-server-side/src/middleware/auth.middleware.ts

```diff
@@ -0,0 +1,26 @@
+import { NextFunction, Request, Response } from "express";
+import jwt from "jsonwebtoken";
+
+export async function authMiddleware(req: Request, res: Response, next: NextFunction) {
+  const authHeader = req.headers.authorization;
+
+  if (!authHeader) {
+    res.status(401).send("Unauthorized");
+  } else {
+    const [_, token] = authHeader.split(" ");
+    const { JWT_SECRET_TOKEN = "supersecret" } = process.env;
+
+    const decodedToken = jwt.verify(token, JWT_SECRET_TOKEN);
```

Comment:
Tutaj nie łapiesz wyjątków.

powinno być `try...catch`

Ponieważ jeśli wystąpi błąd np:
```
TokenExpiredError: jwt expired
```

To wywala cały serwer....:
```
[nodemon] app crashed - waiting for file changes before starting...
```

Ten błąd sprowadza się do pewnej złotej reguły @teziovsky.
W zadaniu `e02e04` z Księżniczką jest do tego wprowadzenie.

Powinieneś mieć dostarczony: `globalErrorHanlder`
tak aby być przygotowanym na wszystko.

Przede wszystkim, nie masz jako developer pewności i wiedzy (używajac 3rd party libs) - co rzuci błędem i w jakich okolicznościach.
Dlatego musisz być "nieufny" różnego rodzaju libom i innym tworom.

Nie sposób wszystkiego opakowywać w `try...catch` !
I to prawda, nie ma to nawet sensu.
Jednak sens jest w przygotowaniu - _W KAŻDYM następnym Twoim serwerze_ czegoś takiego:
https://expressjs.com/en/guide/error-handling.html#the-default-error-handler

Takie coś powinno być jako ostatni `middleware` wpisane w `server.ts`

Reply:
W tym układzie zamiast `try...catch` dostarczasz middleware - którego zadaniem jest obsługa błędów nieobsłużonych przez inne elementy serwera.

Dzieki zastosowaniu klas i w tym wypadku: `JsonWebTokenError` jesteś w stanie określić - kto rzucił błąd w środku swojego `errorHandler`'a.